core/overdrive.py

```diff
@@ -231,31 +231,43 @@ def __init__(self, _db: Session, collection: Collection):
         self._external_integration = collection.external_integration
         self._collection_id = collection.id
 
+        # Initialize configuration information.
+        self._configuration_storage = ConfigurationStorage(self)
+        self._configuration_factory = ConfigurationFactory()
+        self._configuration = OverdriveConfiguration(
+            configuration_storage=self._configuration_storage, db=_db
+        )
+
         if collection.parent:
             # This is an Overdrive Advantage account.
             self.parent_library_id = collection.parent.external_account_id
 
             # We're going to inherit all of the Overdrive credentials
             # from the parent (the main Overdrive account), except for the
             # library ID, which we already set.
-            collection = collection.parent
+            parent_integration = collection.parent.external_integration
+
+            self._configuration.client_key = parent_integration.setting(
+                "overdrive_client_key"
+            )
+            self._configuration.client_password = parent_integration.setting(
+                "overdrive_client_secret"
+            )
+            self._configuration.website_id = parent_integration.setting(
+                "overdrive_website_id"
+            )
         else:
             self.parent_library_id = None
 
-        # Initialize configuration information.
-        self._configuration_storage = ConfigurationStorage(self)
-        self._configuration_factory = ConfigurationFactory()
-        self._configuration = OverdriveConfiguration(
-            configuration_storage=self._configuration_storage, db=_db
-        )
+        if not self._configuration.client_key:
+            raise CannotLoadConfiguration("Overdrive client key is not configured")
+        if not self._configuration.client_password:
+            raise CannotLoadConfiguration(
+                "Overdrive client password/secret is not configured"
+            )
+        if not self._configuration.website_id:
+            raise CannotLoadConfiguration("Overdrive website ID is not configured")
 
-        self._migrate_configuration(
-            collection=collection, configuration=self._configuration
-        )
-        self._server_nickname = (
-            self._configuration.server_nickname
-            or OverdriveConfiguration.PRODUCTION_SERVERS
-        )
```

Comment:
Ah, I think this is why the test broke. Lines 255-258 here are lines I added to include the server family (currently testing or production) in the object. I think you probably wanted to just remove lines 252-254, which invoked the instance's now-removed `_migrate_configuration` method.

Reply:
Oops, yes. Over-zealous removal there.